fvm/src/syscalls/debug.rs

```diff
@@ -21,3 +23,36 @@ pub fn enabled(context: Context<'_, impl Kernel>) -> Result<i32> {
         -1
     })
 }
+
+// TODO: make output path more configurable, maybe add extra gaurds/limitations
+pub fn capture_artifact(
+    context: Context<'_, impl Kernel>,
+    name_off: u32,
+    name_len: u32,
+    data_off: u32,
+    data_len: u32,
+) -> Result<()> {
+    // No-op if disabled.
+    if !context.kernel.debug_enabled() {
+        return Ok(());
+    }
+    let src_dir = std::env::current_dir()
+        .or_fatal()?
+        .canonicalize()
+        .or_fatal()?
+        .join("artifacts");
+    DirBuilder::new()
+        .recursive(true)
+        .create(src_dir.clone())
+        .or_fatal()?;
+
+    let data = context.memory.try_slice(data_off, data_len)?;
+    let name = context.memory.try_slice(name_off, name_len)?;
+    let name = String::from_utf8(name.to_owned())
+        .or_error(fvm_shared::error::ErrorNumber::IllegalArgument)?;
+
+    println!("writing artifact: {} to {:?}", name, src_dir);
+    std::fs::write(src_dir.join(name), data).or_fatal()?;
```

Comment:
We should probably write this to a subdirectory. I'm thinking `$epoch-$randomness/$msg_from_addr/$msg_nonce/$actor_addr/$invocation_count`. Where:

- `$epoch` is the current epoch.
- `$randomness` is derived from tipset randomness (the `get_randomness_from_tickets` syscall but without charging for gas). We should probably invoke this _once_ when constructing the machine to generate the top-level directory. Otherwise, chain reorgs (which replay epochs with different messages) would cause problems.
- `$msg_from_addr` is the message's "origin" address (stored as `origin` in the call manager.
- `$msg_nonce` is the nonce of that message.
- `$actor_addr` is the ID address of the actor saving the artifact.
- `$invocation_count` is an incrementing invocation ID stored in the call manager. Every time we invoke a new actor in the call manager, we increment this ID and store that ID in the kernel. That way, each invocation of an actor in a call stack gets a different directory.

This ensures actor invocations all get their own scoped directories. This isn't really about _security_, it's mostly just to reduce the amount of actor-side logic one might need to avoid clobbering artifacts.